### PR TITLE
EZP-29826: Added switchers component to UI Guidelines

### PR DESCRIPTION
**Pull request by inakijv**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | [EZP-29826](https://jira.ez.no/browse/EZP-29826)
| Versions      | 2.4

Added Switchers component to UI Guidelines Dev Doc section

switchers - ez platform developer documentation

### CODE REVIEW

**Comment by inakijv** [comment]
Ping @DominikaK 

Still have the issue for Switchers with icons with `is-checked` class and the `JS` script (also with Preview switcher). Any suggestions @dew326 @sunpietro @damianz5 ?

**Comment by dew326** [comment]
Are JS files added into the HEAD? Do you know? @damianz5

File: docs/guidelines/components/switchers.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Switchers
+We list here all the switchers that eZ Platform uses.
+
+##<div class="mgt-3 header-line">Switcher</div>
+<div class="ez-guidelines-switcher">
+<p class="mgt-minus-3 mgb-3">We created `switcher` as a SASS @mixin function (check `ezplatform-admin-ui/src/bundle/Resources/public/scss/_mixins.scss`).</p>
+<p  class="mgb-5">Specify its size in the @mixin function: `@include checkbox-switcher($size);` in the corresponding `.scss` stylesheet. By default its size defined is 2rem.</p>
```
Comment (DominikaK) [inline docs/guidelines/components/switchers.md]: `By default its size is set to 2rem.`

File: docs/guidelines/components/switchers.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# Switchers
+We list here all the switchers that eZ Platform uses.
+
+##<div class="mgt-3 header-line">Switcher</div>
+<div class="ez-guidelines-switcher">
+<p class="mgt-minus-3 mgb-3">We created `switcher` as a SASS @mixin function (check `ezplatform-admin-ui/src/bundle/Resources/public/scss/_mixins.scss`).</p>
```
Comment (DominikaK) [inline docs/guidelines/components/switchers.md]: Could we have a link here instead of just a path?

File: docs/css/ez-guidelines-switchers.css [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+.ez-guidelines .ez-guidelines-switcher-icons .ez-checkbox-icon.ez-page-info-bar-switcher {
+    border: 2px solid #f15a10;
+}
+.ez-guidelines .ez-guidelines-switcher-icons .ez-checkbox-icon.ez-page-info-bar-switcher:before {
+    background-color: #f15a10;
+}
+.ez-guidelines .ez-guidelines-switcher-icons .ez-checkbox-icon.is-checked.ez-page-info-bar-switcher .ez-icon + .ez-icon {
+    fill: #f15a10;
+}
```
Comment (damianz5) [inline docs/css/ez-guidelines-switchers.css]: add new rule-set here:
```
.ez-guidelines .ez-guidelines-switcher-icons .ez-checkbox-icon.ez-page-info-bar-switcher:not(.is-checked) .ez-icon.ez-icon--view {
    fill: #f15a10;
}
```

File: docs/js/ez-guidelines-switchers.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+(function(global, doc) {
```
Comment (damianz5) [inline docs/js/ez-guidelines-switchers.js]: try this one:
``` js
(function(global, doc) {
    const checkboxIcons = doc.querySelectorAll(".ez-checkbox-icon");
    const toggleChecked = event => {
        event.currentTarget.classList.toggle("is-checked");
        event.preventDefault();
    };

    checkboxIcons.forEach(element => element.addEventListener("click", toggleChecked, false));

    doc.querySelectorAll(".ez-preview-switcher").forEach(element => {
        const CLASS_PREVIEW_ACTION_SELECTED = 'ez-preview-switcher__action--selected';
        const previewActions = [...element.querySelectorAll('.ez-preview-switcher__action')];
        previewActions.forEach(element => element.addEventListener("click", event => {
            previewActions.forEach((action) => action.classList.remove(CLASS_PREVIEW_ACTION_SELECTED));
            event.currentTarget.classList.add(CLASS_PREVIEW_ACTION_SELECTED);
        }, false));
    }, false);
})(window, window.document);
```

**Comment by inakijv** [comment]
Awesome @damianz5 Thanks!
Ping @dew326 & @sunpietro ready for Dev review now.

**Pull request merged**